java/ql/lib/semmle/code/java/frameworks/apache/Codec.qll

```diff
@@ -0,0 +1,135 @@
+        "org.apache.commons.codec.language;Nysiis;true;nysiis;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language;RefinedSoundex;true;RefinedSoundex;(String);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;RefinedSoundex;true;RefinedSoundex;(char[]);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;RefinedSoundex;true;soundex;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language;Soundex;true;Soundex;(String);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;Soundex;true;Soundex;(String,boolean);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;Soundex;true;Soundex;(char[]);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;Soundex;true;soundex;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;BCodec;true;BCodec;(Charset);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.net;BCodec;true;BCodec;(Charset,CodecPolicy);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.net;BCodec;true;encode;(String,Charset);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;BCodec;true;encode;(String,Charset);;Argument[1];ReturnValue;taint;",
+        "org.apache.commons.codec.net;BCodec;true;encode;(String,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;BCodec;true;getCharset;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.net;QCodec;true;QCodec;(Charset);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.net;QCodec;true;encode;(String,Charset);;Argument[1];ReturnValue;taint;",
+        "org.apache.commons.codec.net;QCodec;true;encode;(String,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;QCodec;true;getCharset;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.net;QuotedPrintableCodec;true;QuotedPrintableCodec;(Charset);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.net;QuotedPrintableCodec;true;QuotedPrintableCodec;(Charset,boolean);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.net;QuotedPrintableCodec;true;decode;(String,Charset);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;QuotedPrintableCodec;true;decode;(String,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;QuotedPrintableCodec;true;encode;(String,Charset);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;QuotedPrintableCodec;true;encode;(String,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;QuotedPrintableCodec;true;getCharset;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.net;URLCodec;true;URLCodec;(String);;Argument[0];Argument[-1];taint;",
```

Comment:
Same thing: `Charset` probably doesn't carry taint, and there are other `encode` and `decode` methods worth modeling.

Reply:
Same as above